src/main/java/com/uid2/admin/vertx/service/SharingService.java

```diff
@@ -15,82 +15,227 @@
 import org.slf4j.LoggerFactory;
 import org.slf4j.Logger;
 
+import java.time.Instant;
 import java.util.*;
 import java.util.stream.Collectors;
 
 public class SharingService implements IService {
     private final AuthMiddleware auth;
 
     private final WriteLock writeLock;
-    private final KeyAclStoreWriter storeWriter;
-    private final RotatingKeyAclProvider keyAclProvider;
+    private final KeysetStoreWriter storeWriter;
+    private final RotatingKeysetProvider keysetProvider;
+    private final IKeysetKeyManager keyManager;
     private static final Logger LOGGER = LoggerFactory.getLogger(SharingService.class);
 
     public SharingService(AuthMiddleware auth,
                           WriteLock writeLock,
-                          KeyAclStoreWriter storeWriter,
-                          RotatingKeyAclProvider keyAclProvider) {
+                          KeysetStoreWriter storeWriter,
+                          RotatingKeysetProvider keysetProvider,
+                          IKeysetKeyManager keyManager) {
         this.auth = auth;
         this.writeLock = writeLock;
         this.storeWriter = storeWriter;
-        this.keyAclProvider = keyAclProvider;
+        this.keysetProvider = keysetProvider;
+        this.keyManager = keyManager;
     }
 
     @Override
     public void setupRoutes(Router router) {
         router.get("/api/sharing/lists").handler(
-                auth.handle(this::handleKeyAclListAll, Role.SHARING_PORTAL)
+                auth.handle(this::handleAllowlistListALL, Role.SHARING_PORTAL)
         );
         router.get("/api/sharing/list/:siteId").handler(
-                auth.handle(this::handleKeyAclList, Role.SHARING_PORTAL)
+                auth.handle(this::handleAllowlistList, Role.SHARING_PORTAL)
         );
         router.post("/api/sharing/list/:siteId").handler(
-                auth.handle(this::handleKeyAclSet, Role.SHARING_PORTAL)
+                auth.handle(this::handleAllowlistSet, Role.SHARING_PORTAL)
```

Comment:
Good stuff that we do semantic POST/GET convention!
Was quite confused on the handler methods though - `handleAllowlistList` and `handleAllowlistSet` sound like they'll return allowlists as a list or as a set. Perhaps we could rename them to something like `handleSetAllowlist` or even better - `setAllowlist`?

Reply:
Also is it still reasonable to have those consistently be `GET /api/sharing/lists`, `GET /api/sharing/lists/:siteId` and `POST /api/sharing/lists/:siteId`? Now they're not consistent with standard HTTP REST conventions.

Reply:
Also Handle is kind of our convention from the other API, so I am going to preserve it here.